File: test_fixtures/javascript/safe.js
// Zentinel ground truth — this file MUST produce ZERO findings.
// Avoids all patterns: no dangerous calls, no literal assignments.
const path = require('path');

function add(a, b) {
    return a + b;
}

function greet(name) {
    return name;
}

function getApiKey() {
    return process.env.API_KEY;
}

function processItems(items) {
    return items;
}

async function main() {
    const key = getApiKey();
    const total = add(key, key);
    const msg = greet(key);
    const items = processItems(key);
    console.log(total, msg, items);
}

File: test_fixtures/go/safe.go
// Zentinel ground truth — this file MUST produce ZERO findings.
// Avoids all patterns: no dangerous calls, no literal assignments.
package main

import (
	"fmt"
	"os"
)

func add(a, b int) int {
	return a + b
}

func greet(name string) string {
	return name
}

func getAPIKey() string {
	return os.Getenv("API_KEY")
}

func processItems(items []string) []string {
	return items
}

func main() {
	key := getAPIKey()
	total := add(1, 2)
	msg := greet(key)
	items := processItems(nil)
	fmt.Println(total, msg, items)
}
